scripts/deploy.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const fs = require('fs');
+const execFile = require('child_process').execFile;
+const S3 = require('aws-sdk/clients/s3');
+const pkg = require('../package');
+
+const bucket = new S3({
+  params: {
+    Bucket: 'www.mapbox.com/assembly/' + pkg.version
+  }
+});
+
+function upload(params) {
+  return new Promise((resolve, reject) => {
+    bucket.upload(params, (err) => {
```

Comment:
The return value of `s3.upload` has [a `promise` property](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3/ManagedUpload.html#promise-property) that makes obviates the need for this wrapper function.